File: 07-03-viewmodel/java-07-03-viewmodel/app/src/main/java/com/example/viewmodel/MainActivity.java
package com.example.viewmodel;

import android.os.Bundle;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        TextView textView = findViewById(R.id.textView);
        PokeViewModel pokeViewModel = new ViewModelProvider(this).get(PokeViewModel.class);
        pokeViewModel.getResults().observe(this, results -> {
            String text = "";
            for (Result result : results) {
                text += result.name + "\n";
            }
            textView.setText(text);
        });
    }
}

File: 07-03-viewmodel/kotlin-07-03-viewmodel/app/src/main/java/com/example/viewmodel/MainActivity.kt
package com.example.viewmodel

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val textView = findViewById<TextView>(R.id.textView)
        val pokeViewModel = ViewModelProvider(this).get<PokeViewModel>(PokeViewModel::class.java)
        pokeViewModel.results.observe(this, Observer { results ->
            var text = ""
            if (results != null) {
                for (result in results) {
                    text += result.name + "\n"
                }
                textView.text = text
            }
        })
    }
}
